### PR TITLE
feat: add aria-disabled to button component

### PR DESCRIPTION
**Pull request by chida09**
ref #1737

## Description

Added aria-disabled to the button component and updated the test.

## Is this a breaking change (Yes/No):

No

## Additional Information
Due to other requirements, the issue will not be closed after the merge of this PR.

### CODE REVIEW

**Review by 108yen — CHANGES_REQUESTED** [review]
Please chack quality.

File: packages/components/button/tests/button.test.tsx [hunk 1]
```diff
@@ -105,6 +105,24 @@ describe("<Button />", () => {
...
+    rerender(
+      <Button as="div" data-testid="btn">
+        Valid Button
+      </Button>,
+    )
+
+    const buttonAsDiv = getByTestId("btn")
+    expect(buttonAsDiv).not.toHaveAttribute("disabled")
+    expect(buttonAsDiv).not.toHaveAttribute("aria-disabled")
+  })
```
Comment (108yen) [inline packages/components/button/tests/button.test.tsx:124]: This test is unnecessary.

**Review by 108yen — COMMENTED** [review]
I don't see anything in the issue to add `aria-disabled`, is this change correct?
Have you discussed this with the accessibility team?

File: packages/components/button/tests/button.test.tsx (hunk 1 above)
Comment (chida09) [reply packages/components/button/tests/button.test.tsx:124]: Why is this test code considered unnecessary?

Comment (chida09) [reply packages/components/button/tests/button.test.tsx:124]: @108yen 

>Disabled State:
 When the action associated with a button is unavailable, the button has aria-disabled="true".

Isn't this included in the requirements?
I understood that we were supposed to address the items with unchecked checkboxes.

Comment (108yen) [reply packages/components/button/tests/button.test.tsx:124]: @chida09 
I think `aria-disabled` tests are covered in one test below.
If I am mistaken, please point it out.

> Why is this test code considered unnecessary?
I think it's a good idea to explicitly test that the button is enabled.

Sorry, I misunderstood that the checked items were the ones that should be corrected!

> Isn't this included in the requirements?
> I understood that we were supposed to address the items with unchecked checkboxes.
> 
> In our team, I've been told to work on issues that seem feasible to start with.

Comment (chida09) [reply packages/components/button/tests/button.test.tsx:124]: @108yen 
>I think aria-disabled tests are covered in one test below.
If I am mistaken, please point it out.

Of course, adding a test case for the presence of aria-disabled when the button is disabled would satisfy the minimum requirement.
It seems we might have different approaches to test coverage. My intention was to create explicit test cases for both the presence of aria-disabled when the button is disabled and its absence when it is enabled.

Since this may be a matter of team policy, I will proceed as per your review, understanding that no additional test cases are needed at this time.

>Sorry, I misunderstood that the checked items were the ones that should be corrected!

I see. Thanks for checking and confirming.

**Review by 108yen — DISMISSED** [review]
Thanks for the explanation!
If you are implementing the “add aria-disabled only if the role is a button” feature in the button component, then I think you are right to add that test.

Approved.

**Comment by hirotomoyamada** [comment]
@chida09 @setodeve 

Hi, thanks for the PR.

When you set `disabled` on a button element, the `aria-disabled` does not read. Is this really a meaningful fix?

- https://www.reddit.com/r/reactjs/comments/18hc7pz/should_you_use_ariadisabled_or_disabled_for/
- https://css-tricks.com/making-disabled-buttons-more-inclusive/

In the process of correcting this, did you look at these URL articles and others?

I feel that this revision lacks discussion and thought.

**Comment by chida09** [comment]
@hirotomoyamada 
Thank you for your feedback.
I'll review the changes and submit for your review again.

**Comment by chida09** [comment]
As few other libraries have adopted aria-disabled for buttons, we will close this issue as not requiring further action.

https://react-spectrum.adobe.com/react-spectrum/Button.html#disabled
https://mui.com/base-ui/react-button/
https://ant.design/components/button
